### DISCUSSION TITLE
[Feature Added] Maybe impl uv_ip_name to get name from sockaddr?

### DISCUSSION THREAD

**Discussion by duskmoon314** [libuv/libuv#3365] [Ideas]
Hello. I'm learning to use libuv and trying to build a udp server that supports both IPv4 and IPv6. I am using `uv_getaddrinfo` to create a socket from args passed from the shell. Since I want to support both IPv4 and IPv6, I use a `switch` with `uv_ip4_name` and `uv_ip6_name` to get name from sockaddr.

I wonder whether it is good to impl `uv_ip_name` to get name from sockaddr. Maybe code like this:

```c
int uv_ip_name(const struct sockaddr *src, char *dst, size_t size) {
  switch (src->sa_family) {
  case AF_INET:
    return uv_inet_ntop(AF_INET, &((struct sockaddr_in *)src)->sin_addr, dst,
                        size);
  case AF_INET6:
    return uv_inet_ntop(AF_INET6, &((struct sockaddr_in6 *)src)->sin6_addr, dst,
                        size);
  default:
    return UV_EAFNOSUPPORT;
  }
}
```

**Comment by bnoordhuis** [comment]
Seems useful, no objections from me. Want to open a PR? Please include docs and tests.

**Reply by duskmoon314** [reply]
Sure!

**Reply by bnoordhuis** [reply]
For posterity: added in #3368.
